### PR TITLE
Fix 403 Bad Authentication with newer urllib3.

### LINKED ISSUE #24: {'Error': 'BadAuthentication'}
I keep getting "{'Error': 'BadAuthentication'}"  with either account password with 2factor authentication disabled and app password with 2fa enabled.

### PR DESCRIPTION
**Pull request by jkuebart**
Closes #24.

With the analysis in urllib3/urllib3#2101 I was able to find a workaround. It works for me now up to the current version urllib3-1.26.3.

### CODE REVIEW

File: gpsoauth/__init__.py [hunk 1]
```diff
@@ -16,9 +18,50 @@
...
+        "ECDH+AES",
+        "DH+AES",
+        "RSA+AESGCM",
+        "RSA+AES",
+        "!aNULL",
+        "!eNULL",
+        "!MD5",
+        "!DSS",
+    ]
+)
```
Comment (simon-weber) [inline gpsoauth/__init__.py:37]: this was pulled from [here](https://github.com/urllib3/urllib3/blob/e47d3e7456b95c59ba85c55b6af0c898b63e5c4f/src/urllib3/util/ssl_.py#L64-L99), yeah? Would it be possible to build our list from theirs and just maintain a blocklist on our end for the known troublesome ciphers? I'm worried about having to maintain this list.

Comment (jkuebart) [reply gpsoauth/__init__.py:37]: Yes, that's the correct source. I had to remove the last entry `!AESCCM` which was added recently to make it work again. The rest of the list can stay the same for now.

Comment (jkuebart) [reply gpsoauth/__init__.py:37]: so  something like
```
   context.set_ciphers(cipher for cipher in urllib3.util.ssl_.DEFAULT_CIPHERS if cipher != "!AESCCM")
```
should also work.

Comment (jkuebart) [reply gpsoauth/__init__.py:37]: let me know if I should test & update this PR

Comment (simon-weber) [reply gpsoauth/__init__.py:37]: great; that sounds good to me 👍

**Comment by simon-weber** [comment]
thanks! planning to give this a quick test on my end later today, and if it looks good I'll cut a new release.

**Comment by DurgNomis-drol** [comment]
I can also confirm that it fixes the problem

**Pull request merged**